doc: Fill out the missing information about FEMs (#8620)
nrfconnect/sdk-nrf: doc/nrf/ug_radio_fem.rst

   The MPSL library provides API to configure FEM.
   See :ref:`nrfxlib:mpsl_lib` in the nrfxlib documentation for details.
#. Enable support for MPSL implementation in |NCS| by setting the :kconfig:option:`CONFIG_MPSL` Kconfig option to ``y``.
#. Enable support for the FEM subsystem in |NCS| by setting the :kconfig:option:`CONFIG_MPSL_FEM` Kconfig option to ``y``.

> #. Choose the used FEM implementation by selecting an appropriate Kconfig option.

Reviewer: ```suggestion
#. Choose the used FEM implementation by selecting the appropriate Kconfig option.
```

The following FEM implementations are supported:

* The nRF21540 GPIO implementation, see :ref:`ug_radio_fem_nrf21540_gpio`.
  To use it, set the :kconfig:option:`CONFIG_MPSL_FEM_NRF21540_GPIO` Kconfig option to ``y``.
* The nRF21540 GPIO SPI implementation, see :ref:`ug_radio_fem_nrf21540_spi_gpio`.
  To use it, set the :kconfig:option:`CONFIG_MPSL_FEM_NRF21540_GPIO_SPI` Kconfig option to ``y``.
* The nRF21540 2-pin simple GPIO implementation.
  To use it, set the :kconfig:option:`CONFIG_MPSL_FEM_SIMPLE_GPIO` Kconfig option to ``y``.

Setting the FEM output power
----------------------------

> The FEM gain to use by the simple GPIO FEM implementation is provided by the ``tx_gain_db`` property in devicetree and must represent the real gain of the FEM.

Reviewer 2: typo: devicetree
device tree

Reviewer: The spec name is "devicetree", I'd leave it as is (https://www.devicetree.org/)

The property must represent the real gain of the FEM.
This implementation does not support controlling the gain value during runtime.

> nRF21540 implementations have the gain set to 10 by default.

Reviewer: ```suggestion
nRF21540 implementations have the gain set to ``10`` by default.
```

You can set a different gain value to use through the :kconfig:option:`CONFIG_MPSL_FEM_NRF21540_TX_GAIN_DB`  option, but it has to match the value of one of the POUTA (:kconfig:option:`CONFIG_MPSL_FEM_NRF21540_TX_GAIN_DB_POUTA` ) or POUTB (:kconfig:option:`CONFIG_MPSL_FEM_NRF21540_TX_GAIN_DB_POUTB`) gains.

.. caution::

>    :kconfig:option:`CONFIG_MPSL_FEM_NRF21540_TX_GAIN_DB_POUTA` and :kconfig:option:`CONFIG_MPSL_FEM_NRF21540_TX_GAIN_DB_POUTB` are by default set to 20 and 10 and these are factory-precalibrated gain values.

Reviewer: ```suggestion
   :kconfig:option:`CONFIG_MPSL_FEM_NRF21540_TX_GAIN_DB_POUTA` and :kconfig:option:`CONFIG_MPSL_FEM_NRF21540_TX_GAIN_DB_POUTB` are by default set to ``20`` and ``10`` and these are factory-precalibrated gain values.
```

The user should not change these values, unless POUTA and POUTB were calibrated to different values on special user request.

> Runtime control of the gain can be enabled, by setting the :kconfig:option:`CONFIG_MPSL_FEM_NRF21540_RUNTIME_PA_GAIN_CONTROL` to y.

Reviewer: ```suggestion
To enable runtime control of the gain, set the :kconfig:option:`CONFIG_MPSL_FEM_NRF21540_RUNTIME_PA_GAIN_CONTROL` to ``y``.
```

> This option causes that the gain of the FEM is adjusted dynamically during run-time depending on the power requested by a protocol driver for each individual transmission.

Reviewer: ```suggestion
This option makes the gain of the FEM to be adjusted dynamically during runtime, depending on the power requested by the protocol driver for each transmission.
```

For the nRF21540 GPIO implementation, you must enable the **MODE** pin in devicetree.

> For the nRF21540 GPIO SPI implementation, no additional configuration is needed as the gain setting is transmitted over the SPI bus to nRF21540.

Reviewer: ```suggestion
For the nRF21540 GPIO SPI implementation, no additional configuration is needed as the gain setting is transmitted over the SPI bus to the nRF21540.
```

> Runtime gain control of the gain can be enabled, by setting the :kconfig:option:`CONFIG_MPSL_FEM_NRF21540_RUNTIME_PA_GAIN_CONTROL` to y.
> This will cause MPSL to switch the gain between the values POUTA and POUTB, attempting to minimize signal distortions.
> When using the nRF21540 GPIO implementation, the MODE pin must be enabled in devicetree.
> This isn't needed for the GPIO SPI implementation, as the implementation uses the SPI gain instead of the mode pin in order to control the gain.

Reviewer 3: I don't get `This isn't needed for the GPIO SPI implementation` the `CONFIG_MPSL_FEM_NRF21540_RUNTIME_PA_GAIN_CONTROL` is not needed?

What about:
```suggestion
Runtime gain control of the gain can be enabled, by setting the :kconfig:option:`CONFIG_MPSL_FEM_NRF21540_RUNTIME_PA_GAIN_CONTROL` to y.
This option causes that the gain of the FEM is adjusted dynamically during run-time depending on power requested by a protocol driver for each individual transmission. 
For the nRF21540 GPIO implementation the MODE pin must be enabled in device tree. 
For the nRF21540 GPIO SPI implementation no additional configuration is needed as the gain setting is transmitted over the SPI bus to nRF21540.
```

> For the nRF21540 GPIO implementation the MODE pin must be enabled in device tree.
> For the nRF21540 GPIO SPI implementation no additional configuration is needed as the gain setting is transmitted over the SPI bus to nRF21540.

Reviewer: ```suggestion
For the nRF21540 GPIO implementation, you must enable the **MODE** pin in devicetree.
For the nRF21540 GPIO SPI implementation, no additional configuration is needed as the gain setting is transmitted over the SPI bus to nRF21540.
```

You can use only the :ref:`nrfxlib:mpsl_fem` API if your application does not require other MPSL features.

> This might be useful when you want to run simple radio protocols that are not intended to be used concurrently with other protocols.

Reviewer 2: Line: 105:
Suggest change: `...given a real transmission power at the device output...` What do you think?

Reviewer 2: Line: 106:
Could we describe more examples and add also internal condition to it?
...external conditions such as temperature... -> external or internal conditions (for example temperature, power supply voltage, channel frequency) might affect...

Author: I deleted the "the" word, it was there by accident. However I prefer not to write "real transmission power", as it might not be the truth (the RF path may cause attenuation)

Author: Regarding the second comment - I'd prefer not to introduce changes to the part about the model, as the focus of this PR is different.
